src/main/java/net/dries007/tfc/objects/te/TEFirePit.java

```diff
@@ -322,7 +327,57 @@ private void handleInputMelting(ItemStack stack)
                 outputStack = inventory.insertItem(SLOT_OUTPUT_1, outputStack, false);
                 if (!outputStack.isEmpty())
                 {
-                    inventory.insertItem(SLOT_OUTPUT_2, outputStack, false);
+                    outputStack = inventory.insertItem(SLOT_OUTPUT_2, outputStack, false);
+                }
+                if (!outputStack.isEmpty()) // Couldn't merge directly
+                {
+                    // If both the output and input is the same food, try merging-updating the creation date to the earliest one
```

Comment:
Two concerns:
 - the "merge two stacks that have different dates" should become a helper method in `CapabilityFood`. It's already used in the crafting recipe for food merging.
 - Do not try and mix traits / force stacks to stack if their traits don't match. The only thing that should be forcibly stacked is creation date